controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -301,10 +306,137 @@ func (r *ReplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	err = r.processLastActionResult(ctx, localRG, remoteClient, log)
+	if err != nil {
+		r.EventRecorder.Eventf(localRG, eventTypeWarning, eventReasonUpdated,
+			"failed to process the last action %s", localRG.Status.LastAction.Condition)
+	}
+
 	log.V(common.InfoLevel).Info("RG has already been synced to the remote cluster")
 	return ctrl.Result{}, nil
 }
 
+func (r *ReplicationGroupReconciler) processLastActionResult(ctx context.Context, group *storagev1alpha1.DellCSIReplicationGroup, remoteClient connection.RemoteClusterClient, log logr.Logger) error {
+	if len(group.Status.Conditions) == 0 || group.Status.LastAction.Time == nil {
+		log.V(common.InfoLevel).Info("No action to process")
+		return nil
+	}
+
+	if strings.Contains(group.Status.LastAction.Condition, "CREATE_SNAPSHOT") {
```

Comment:
What happens if the last action (CREATE_SNAPSHOT) actually failed? Will that still be processed by the replication controller?

```
lastAction:
    condition: Action RESUME succeeded
    time: "2023-02-08T20:52:02Z"
```

Reply:
Added check to not continue if an error message exists.